### ISSUE TITLE
Backport blacklist changes from 3.0

### ISSUE THREAD

**Issue by dpryan79** [deeptools/deepTools#652]
From the sound of things it might be good to backport the version 3.0 behavior of exiting with an error if the blacklist contains overlapping regions to version 2.5.

@steffenheyne @thomasmanke @mirax87: Thoughts? The only thing holding up release of deepTools 3.0 is that it depends on the next pysam release.

**Comment by fidelram** [comment]
I will support backporting. This is an important update.

**Comment by steffenheyne** [comment]
Me too! 
I would also like to have a warning/log output of the minimum distance of regions seen in the blacklist. Maybe also a short text message that depending on the exact fragment sizes, double counting can still occur. 

The documentation text could be also more descriptive here to understand possible issues better.

**Comment by dpryan79** [comment]
@steffenheyne Can do, though that'll take an hour or two to test and implement. The fragment size isn't known when blacklisting is done, so I'll just issue a warning if the minimum distance is less than 1kb.

**Comment by steffenheyne** [comment]
Great, Thanks!

**Comment by mirax87** [comment]
I agree as well. Regarding the response strategy, I can picture a stronger measure than a warning (easy to miss e.g. in workflows). 
My suggestion would be to throw an error. If the user still wants to use that blacklist (after e.g. manual inspection and confirmation), the user could activate a flag like "--ignoreBlacklistWarnings".

**Comment by dpryan79** [comment]
Travis is testing this now.

@mirax87 I'm trying to avoid needing to pass more arguments through 5 or more layers of function calls (not to mention that adding options in a bugfix is frowned upon).

@steffenheyne The documentation is being updated a bit. Please have a look at the hotfix/2.5.5 branch.

**Comment by dpryan79** [comment]
Given the backlog on Travis, this will probably take another 12-16 hours before it's released (and then another day or two for conda). Note that the previously mentioned risk of double counting given neighboring blacklist regions is over-stated. I checked how I had coded this and it shouldn't be possible to double count a read unless it's fully contained within multiple blacklisted regions (i.e., they do have to overlap). Having said that, it doesn't make any biological sense to have neighboring blacklisted regions, they should get merged together. So I'll just tweak the warning message to mention that instead (I'll update the documentation a bit too).
